pkg/cache/snapshot.go

```diff
@@ -133,6 +132,14 @@ func (c *clusterQueue) snapshot() *ClusterQueueSnapshot {
 		Quotas:                        c.quotas,
 	}
 
+	for _, rg := range c.ResourceGroups {
+		cc.ResourceGroups = append(cc.ResourceGroups, ResourceGroup{
+			CoveredResources: rg.CoveredResources.Clone(),
```

Comment:
nit: I don't think either of these Clones are necessary - when we update ResourceGroup, we simply overwrite these fields with new object, never mutate the existing ones.

Reply:
on the other hand, making a copy is cheap... ignore above comment please